src/agent/Core/SpawningKit/Handshake/Perform.h

```diff
@@ -196,8 +196,18 @@ class HandshakePerform {
 
 		while (true) {
 			unsigned long long timeout = 100000;
+			bool pingable;
 
-			if (pingTcpServer("127.0.0.1", session.expectedStartPort, &timeout)) {
+			try {
+				pingable = pingTcpServer("127.0.0.1", session.expectedStartPort, &timeout);
+			} catch (const std::exception &e) {
+				P_WARN("Error checking whether 127.0.0.1:" << session.expectedStartPort
+					<< " is connectable: " << e.what());
+				syscalls::usleep(50000);
```

Comment:
this should either be configurable or much less than 50s

Reply:
This is 50 milliseconds, not 50 seconds. usleep is microseconds.